tests/testthat/custom_tests/test-regional-dataset.R

```diff
@@ -38,23 +38,37 @@ test_regional_dataset <- function(source, level, download = FALSE) {
   }
 
   if (download) {
-    test_that(paste0(data_name, " downloads sucessfully"), {
+    test_that(paste0(data_name, " downloads successfully"), {
       region$download()
       purrr::walk(region$data$raw, function(data) {
         expect_s3_class(data, "data.frame")
         expect_true(nrow(data) > 0)
         expect_true(ncol(data) >= 2)
       })
     })
+  } else {
+    region$data$raw <- readRDS(raw_path)
+  }
+
+  if (vroom_check & download) {
+    problems_list <- purrr::map(region$data$raw,
+                                vroom::problems)
+    purrr::walk(problems_list, function(data) {
+      expect_s3_class(data, "data.frame")
+      expect_true(nrow(data) == 0)
+      # if( nrow(data) > 0) {
+      #   warning(data)
+      # }
+    })
+  }
+
+  if (download) {
```

Comment:
Is there a reason this section goes after the vroom check? I think if we are only testing vroom when download is on (which is fine) then your logic from before makes more sense. Its only if we want to test vroom apart from downloading (i.e on the snapshot) that we would want to change this up.

Reply:
I wanted to be able to run the `vroom::problems` call on the full table, not curtailed. It appears that `vroom::problems` doesn't work on the table when loaded from the file, though it does work on the curtailed version. I've not dug deep, but I think that `problems` just shows a tibble which is hidden data on the frame offered by `vroom`, but that the link to this data (or the data itself) is lost or broken when writing to file.

Reply:
ah interesting that makes sense though quite surprising as `saveRDS` is usually pretty good about preserving attributes.